tensorboard/plugins/scalar/tf_scalar_dashboard/tf-scalar-card.html

```diff
@@ -314,11 +316,13 @@
       },
       _updateDownloadLink() {
         const svgStr = this.$$('tf-line-chart-data-loader').exportAsSvgString();
-        // The SVG code string may include hash characters, such as an
-        // attribute `clipPath="url(#foo)"`. Thus, we base64-encode the
-        // data so that such a hash is not interpreted as a fragment
-        // specifier, truncating the SVG. (See issue #1874.)
-        this.$$('#svgLink').href = `data:image/svg+xml;base64,${btoa(svgStr)}`;
+        const bytes = new Uint8Array(svgStr.length).map((_, i) => svgStr.charCodeAt(i));
+        const blob = new Blob([bytes], {type: 'image/svg+xml'});
+        this.$$('#svgLink').setAttribute('href', URL.createObjectURL(blob));
+      },
+      _removeDownloadLink() {
+        URL.revokeObjectURL(this.$$('#svgLink').href);
```

Comment:
Is it possible to use `this.$.svgLink` here? Or does that not work due
to the template structure?

Reply:
It does not work because of wrapping `dom-if`.

Reply:
Right, okay. Thanks.

Reply:
For posterity: dom-if (at least in Polymer v1) does not mount the DOM until the first truthy to `if` property. As a result, at the shadowRoot creation time (kinda speculating based on experience), the DOM does not exist and thus `this.$` is not populated.